third_party/gtp_ovs/ovs-gtp-patches/2.15/0022-Add-GTP-IPv6-support.patch

```diff
@@ -0,0 +1,1405 @@
++ n_packets=3, n_bytes=354, priority=100,ipv6,ipv6_dst=2001::1 actions=output:1
++ n_packets=1, n_bytes=63, priority=100,ipv6,ipv6_dst=2001::2 actions=drop
++])
++
++dnl sleep 1000
++OVS_TRAFFIC_VSWITCHD_STOP
++AT_CLEANUP
++
++AT_SETUP([layer3 - IPv6 packets over IPv4 GTP])
++OVS_TRAFFIC_VSWITCHD_START([set Bridge br0 other-config:hwaddr="00:12:34:56:78:bb"])
++OVS_CHECK_GTP_L3()
++OVS_CHECK_MIN_KERNEL(4,10)
++
++ADD_BR([br-underlay])
++
++ADD_NAMESPACES(at_ns0)
++
++dnl Set up underlay link from host into the namespace using veth pair.
++ADD_VETH(p0, at_ns0, br-underlay, "172.31.1.1/24")
++AT_CHECK([ip addr add dev br-underlay "172.31.1.100/24"])
++AT_CHECK([ip link set dev br-underlay up])
++
++AT_CHECK([modprobe vport_gtp])
++
++dnl Set up tunnel endpoints on OVS outside the namespace and with a native
++dnl linux device inside the namespace.
++
++ADD_OVS_TUNNEL([gtpu], [br0], [at_gtp0], [172.31.1.1], [2001::2/64])
++AT_CHECK([ip -6 neigh add 2001::1 lladdr 00:12:34:56:78:aa dev br0])
++
++NS_CHECK_EXEC([at_ns0], [ip link set dev lo up])
++NS_CHECK_EXEC([at_ns0], [ip -6 addr add dev lo 2005:cafe::92/64])
++
++AT_CHECK([ovs-ofctl add-flow br-underlay "actions=normal"])
++
++AT_CHECK([ovs-appctl vlog/set dbg], [0], [ignore])
++AT_CHECK([echo 'module openvswitch +p' > /sys/kernel/debug/dynamic_debug/control])
++
++dnl Now add rules for OVS to forward to the tunnel and local port
++AT_CHECK([ovs-ofctl add-flow br0 "priority=1 action=drop"])
++AT_CHECK([ovs-ofctl add-flow br0 "priority=100 ip6,ipv6_dst=2001::1 action=output:at_gtp0"])
++AT_CHECK([ovs-ofctl add-flow br0 "priority=100 ip6,ipv6_dst=2001::2 action=drop"])
++
++AT_CHECK([ovs-ofctl dump-flows br0 | grep "ipv6" | ofctl_strip], [0], [dnl
++ priority=100,ipv6,ipv6_dst=2001::1 actions=output:1
++ priority=100,ipv6,ipv6_dst=2001::2 actions=drop
++])
++
++sleep 1
++NS_CHECK_EXEC([at_ns0], [python3 $srcdir/gtp-packet.py 172.31.1.1 172.31.1.100 2001::1 2001::2 5555 p0 False 1234 2>/dev/null], [0], [dnl
++.
++Sent 1 packets.
++])
++
++AT_CHECK([ovs-ofctl dump-flows br0 | ofctl_strip | grep ipv6 ], [0], [dnl
++ priority=100,ipv6,ipv6_dst=2001::1 actions=output:1
++ n_packets=1, n_bytes=63, priority=100,ipv6,ipv6_dst=2001::2 actions=drop
++])
++
++NS_CHECK_EXEC([at_ns0], [tcpdump -l -n -xx -U ip host 172.31.1.1 -i p0 > ipv6_xmit.pcap  2>/dev/null &])
++
++sleep 1
++AT_CHECK([ping -6 2001::1 -q -c 3 -I br0 2>/dev/null | FORMAT_PING], [0], [dnl
++3 packets transmitted, 0 received, 100% packet loss, time 0ms
++])
++
++OVS_WAIT_UNTIL([cat ipv6_xmit.pcap | egrep "IP 172.31.1.100.2152 > 172.31.1.1.2152: UDP, length 112" 2>&1 1>/dev/null])
++
++AT_CHECK([ovs-ofctl dump-flows br0 | ofctl_strip | grep ipv6 ], [0], [dnl
++ n_packets=3, n_bytes=354, priority=100,ipv6,ipv6_dst=2001::1 actions=output:1
++ n_packets=1, n_bytes=63, priority=100,ipv6,ipv6_dst=2001::2 actions=drop
++])
++
++dnl sleep 1000
++OVS_TRAFFIC_VSWITCHD_STOP
++AT_CLEANUP
++
++
++AT_SETUP([layer3 - GTP IPv4 packets over IPv6 tunnel])
```

Comment:
can you change name to be consistent with above test case to 'IPv4 packets over IPv6 GTP'